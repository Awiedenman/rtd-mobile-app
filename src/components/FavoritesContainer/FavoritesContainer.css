.favorite_card-container {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.favorites-div {
  width: 100%;
  background: #a2a2a2;
  padding: 0 5px 10px 5px;
  height: 230px;
  overflow-y: scroll;
}

.favorites-title-container {
  height: 30px;
  width: 100%;
  background: white;
  display: flex;
  align-items: center;
  justify-content: center;
}

.favorites-title {
  font-size: 17px;
  font-weight: 300;
}