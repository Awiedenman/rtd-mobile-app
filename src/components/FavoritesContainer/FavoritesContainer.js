import React, { Component } from 'react';
import Favorite from '../Favorite/Favorite';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import './FavoritesContainer.css';


export class FavoritesContainer extends Component {
  
  render(){
    const { favorites } = this.props;

    const favoriteIds = favorites.reduce((favoritesObj, favorite) => {
      if (!favoritesObj[favorite.itinerary_id]) {
        favoritesObj[favorite.itinerary_id] = {...favorite};
      }
      return favoritesObj;
    }, {});
    
    const favoritesData = Object.keys(favoriteIds).map((id, index) => {
      return (
        <Favorite 
          favData={favoriteIds[id]}
          key={index}
        />
      );
    });
    return (
      <div className='favorite_card-container'>
        <div className='favorites-div'>
          <h2 className='favorites-title'>Favorites</h2>
          {favoritesData}
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  favorites: state.favorites
});

export default connect(mapStateToProps)(FavoritesContainer);

FavoritesContainer.propTypes = {
  favorites: PropTypes.array
};