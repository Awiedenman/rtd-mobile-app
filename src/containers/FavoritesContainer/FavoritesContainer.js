import React, { Component } from 'react';
import Favorite from '../../components/Favorite/Favorite';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { 
  getFavUrl,
  deleteFavUrl
} from '../../constants/urlGenerator';
import getFavItineraryThunk from '../../thunks/getFavItineraryThunk';
import deleteFavItineraryThunk from '../../thunks/deleteFavItineraryThunk';
import * as routes from '../../constants/routes';
import LoadingPage from '../../components/Loading/Loading';

import './FavoritesContainer.css';


export class FavoritesContainer extends Component {

  searchFavorite = async (event) => {
    const { value } = event.target;
    const { 
      user, 
      getFavItinerary, 
      history 
    } = this.props;
    const url = getFavUrl(user.uid, value);
    await getFavItinerary(url);
    history.push(routes.ITINERARY);
  }

  deleteFavorite = async (event) => {
    const { id } = event.target;
    const {
      user,
      deleteFavItinerary
    } = this.props;
    const fetchObj = deleteFavUrl(user.uid, id);
    await deleteFavItinerary(fetchObj);
  }

  render(){
    const { 
      favorites
    } = this.props;

    const loading = <LoadingPage type='loading-container' />;

    const favoriteIds = favorites.reduce((favoritesObj, favorite) => {
      if (!favoritesObj[favorite.id]) {
        favoritesObj[favorite.id] = {...favorite};
      }
      return favoritesObj;
    }, {});
      
    const favoritesData = Object.keys(favoriteIds).map((id, index) => {
      return (
        <Favorite
          favData={favoriteIds[id]}
          key={index}
          searchFavorite={this.searchFavorite}
          deleteFavorite={this.deleteFavorite}
        />
      );
    });
    
    return (
      <div className='card-container'>
        <div className='card-div'>
          {this.props.isLoading === 'favorite'
            ? loading
            : favoritesData}
        </div>
      </div>
    );
  }

}

export const mapStateToProps = state => ({
  favorites: state.favorites,
  user: state.user,
  isLoading: state.isLoading
});

export const mapDispatchToState = dispatch => ({
  getFavItinerary: url => dispatch(getFavItineraryThunk(url)),
  deleteFavItinerary: fetchObj => dispatch(deleteFavItineraryThunk(fetchObj))
});

export default withRouter(connect(mapStateToProps, mapDispatchToState)(FavoritesContainer));

FavoritesContainer.propTypes = {
  favorites: PropTypes.arrayOf(PropTypes.object),
  user: PropTypes.object,
  getFavItinerary: PropTypes.func,
  deleteFavItinerary: PropTypes.func,
  isLoading: PropTypes.string,
  history: PropTypes.object
};