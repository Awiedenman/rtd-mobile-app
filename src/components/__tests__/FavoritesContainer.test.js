import React from 'react'
import { shallow } from 'enzyme';
import { FavoritesContainer } from '../FavoritesContainer/FavoritesContainer';

describe('FavoritesContainer', () => {
  it('should matchSnapshot', () => {
    const wrapper = shallow(<FavoritesContainer />);
    
    expect(wrapper).toMatchSnapshot();
  });
});
