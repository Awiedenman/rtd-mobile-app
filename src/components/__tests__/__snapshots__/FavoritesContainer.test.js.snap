// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesContainer should matchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoritesContainer
    favorites={
        Array [
            Array [
              Object {
                "arrival_time": "4:11pm",
                "departure_time": "3:51pm",
                "distance": "3.0 mi",
                "duration": "20 mins",
                "end_address": "366 Broadway, Denver, CO 80203, USA",
                "favorite": false,
                "itinerary_id": 7,
                "start_address": "Denver, CO 80202, USA",
                "steps": Array [
                  Object {
                    "arrival_stop": null,
                    "arrival_time": null,
                    "color": null,
                    "credit_name": null,
                    "credit_url": null,
                    "departure_stop": null,
                    "departure_time": null,
                    "distance": "1 ft",
                    "duration": "1 min",
                    "headsign": null,
                    "id": 26,
                    "instructions": "Walk to Union Station",
                    "name": null,
                    "num_stops": null,
                    "possible_route_id": 7,
                    "short_name": null,
                    "travel_mode": "Walking",
                    "vehicle_type": null,
                  },
                  Object {
                    "arrival_stop": "Broadway & 4th Ave",
                    "arrival_time": "4:11pm",
                    "color": "#0277bd",
                    "credit_name": "Regional Transportation District",
                    "credit_url": "http://rtd-denver.com/",
                    "departure_stop": "Union Station",
                    "departure_time": "3:52pm",
                    "distance": "3.0 mi",
                    "duration": "20 mins",
                    "headsign": "Highlands Ranch",
                    "id": 27,
                    "instructions": "Bus towards Highlands Ranch",
                    "name": "South Broadway",
                    "num_stops": 11,
                    "possible_route_id": 7,
                    "short_name": "0",
                    "travel_mode": "TRANSIT",
                    "vehicle_type": "Bus",
                  },
                  Object {
                    "arrival_stop": null,
                    "arrival_time": null,
                    "color": null,
                    "credit_name": null,
                    "credit_url": null,
                    "departure_stop": null,
                    "departure_time": null,
                    "distance": "108 ft",
                    "duration": "1 min",
                    "headsign": null,
                    "id": 28,
                    "instructions": "Walk to 366 Broadway, Denver, CO 80203, USA",
                    "name": null,
                    "num_stops": null,
                    "possible_route_id": 7,
                    "short_name": null,
                    "travel_mode": "Walking",
                    "vehicle_type": null,
                  },
                ],
              },
            ],
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="favorites-div"
>
        <h2
                className="favorites-title"
        >
                Favorites
        </h2>
        <Favorite
                deleteFavorite={[Function]}
                favData={
                        Object {
                                "0": Object {
                                  "arrival_time": "4:11pm",
                                  "departure_time": "3:51pm",
                                  "distance": "3.0 mi",
                                  "duration": "20 mins",
                                  "end_address": "366 Broadway, Denver, CO 80203, USA",
                                  "favorite": false,
                                  "itinerary_id": 7,
                                  "start_address": "Denver, CO 80202, USA",
                                  "steps": Array [
                                    Object {
                                      "arrival_stop": null,
                                      "arrival_time": null,
                                      "color": null,
                                      "credit_name": null,
                                      "credit_url": null,
                                      "departure_stop": null,
                                      "departure_time": null,
                                      "distance": "1 ft",
                                      "duration": "1 min",
                                      "headsign": null,
                                      "id": 26,
                                      "instructions": "Walk to Union Station",
                                      "name": null,
                                      "num_stops": null,
                                      "possible_route_id": 7,
                                      "short_name": null,
                                      "travel_mode": "Walking",
                                      "vehicle_type": null,
                                    },
                                    Object {
                                      "arrival_stop": "Broadway & 4th Ave",
                                      "arrival_time": "4:11pm",
                                      "color": "#0277bd",
                                      "credit_name": "Regional Transportation District",
                                      "credit_url": "http://rtd-denver.com/",
                                      "departure_stop": "Union Station",
                                      "departure_time": "3:52pm",
                                      "distance": "3.0 mi",
                                      "duration": "20 mins",
                                      "headsign": "Highlands Ranch",
                                      "id": 27,
                                      "instructions": "Bus towards Highlands Ranch",
                                      "name": "South Broadway",
                                      "num_stops": 11,
                                      "possible_route_id": 7,
                                      "short_name": "0",
                                      "travel_mode": "TRANSIT",
                                      "vehicle_type": "Bus",
                                    },
                                    Object {
                                      "arrival_stop": null,
                                      "arrival_time": null,
                                      "color": null,
                                      "credit_name": null,
                                      "credit_url": null,
                                      "departure_stop": null,
                                      "departure_time": null,
                                      "distance": "108 ft",
                                      "duration": "1 min",
                                      "headsign": null,
                                      "id": 28,
                                      "instructions": "Walk to 366 Broadway, Denver, CO 80203, USA",
                                      "name": null,
                                      "num_stops": null,
                                      "possible_route_id": 7,
                                      "short_name": null,
                                      "travel_mode": "Walking",
                                      "vehicle_type": null,
                                    },
                                  ],
                                },
                              }
                }
                searchFavorite={[Function]}
        />
</div>,
      "className": "favorite_card-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="favorites-title"
>
            Favorites
</h2>,
          Array [
            <Favorite
              deleteFavorite={[Function]}
              favData={
                            Object {
                                          "0": Object {
                                            "arrival_time": "4:11pm",
                                            "departure_time": "3:51pm",
                                            "distance": "3.0 mi",
                                            "duration": "20 mins",
                                            "end_address": "366 Broadway, Denver, CO 80203, USA",
                                            "favorite": false,
                                            "itinerary_id": 7,
                                            "start_address": "Denver, CO 80202, USA",
                                            "steps": Array [
                                              Object {
                                                "arrival_stop": null,
                                                "arrival_time": null,
                                                "color": null,
                                                "credit_name": null,
                                                "credit_url": null,
                                                "departure_stop": null,
                                                "departure_time": null,
                                                "distance": "1 ft",
                                                "duration": "1 min",
                                                "headsign": null,
                                                "id": 26,
                                                "instructions": "Walk to Union Station",
                                                "name": null,
                                                "num_stops": null,
                                                "possible_route_id": 7,
                                                "short_name": null,
                                                "travel_mode": "Walking",
                                                "vehicle_type": null,
                                              },
                                              Object {
                                                "arrival_stop": "Broadway & 4th Ave",
                                                "arrival_time": "4:11pm",
                                                "color": "#0277bd",
                                                "credit_name": "Regional Transportation District",
                                                "credit_url": "http://rtd-denver.com/",
                                                "departure_stop": "Union Station",
                                                "departure_time": "3:52pm",
                                                "distance": "3.0 mi",
                                                "duration": "20 mins",
                                                "headsign": "Highlands Ranch",
                                                "id": 27,
                                                "instructions": "Bus towards Highlands Ranch",
                                                "name": "South Broadway",
                                                "num_stops": 11,
                                                "possible_route_id": 7,
                                                "short_name": "0",
                                                "travel_mode": "TRANSIT",
                                                "vehicle_type": "Bus",
                                              },
                                              Object {
                                                "arrival_stop": null,
                                                "arrival_time": null,
                                                "color": null,
                                                "credit_name": null,
                                                "credit_url": null,
                                                "departure_stop": null,
                                                "departure_time": null,
                                                "distance": "108 ft",
                                                "duration": "1 min",
                                                "headsign": null,
                                                "id": 28,
                                                "instructions": "Walk to 366 Broadway, Denver, CO 80203, USA",
                                                "name": null,
                                                "num_stops": null,
                                                "possible_route_id": 7,
                                                "short_name": null,
                                                "travel_mode": "Walking",
                                                "vehicle_type": null,
                                              },
                                            ],
                                          },
                                        }
              }
              searchFavorite={[Function]}
/>,
          ],
        ],
        "className": "favorites-div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites",
            "className": "favorites-title",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "deleteFavorite": [Function],
            "favData": Object {
              "0": Object {
                "arrival_time": "4:11pm",
                "departure_time": "3:51pm",
                "distance": "3.0 mi",
                "duration": "20 mins",
                "end_address": "366 Broadway, Denver, CO 80203, USA",
                "favorite": false,
                "itinerary_id": 7,
                "start_address": "Denver, CO 80202, USA",
                "steps": Array [
                  Object {
                    "arrival_stop": null,
                    "arrival_time": null,
                    "color": null,
                    "credit_name": null,
                    "credit_url": null,
                    "departure_stop": null,
                    "departure_time": null,
                    "distance": "1 ft",
                    "duration": "1 min",
                    "headsign": null,
                    "id": 26,
                    "instructions": "Walk to Union Station",
                    "name": null,
                    "num_stops": null,
                    "possible_route_id": 7,
                    "short_name": null,
                    "travel_mode": "Walking",
                    "vehicle_type": null,
                  },
                  Object {
                    "arrival_stop": "Broadway & 4th Ave",
                    "arrival_time": "4:11pm",
                    "color": "#0277bd",
                    "credit_name": "Regional Transportation District",
                    "credit_url": "http://rtd-denver.com/",
                    "departure_stop": "Union Station",
                    "departure_time": "3:52pm",
                    "distance": "3.0 mi",
                    "duration": "20 mins",
                    "headsign": "Highlands Ranch",
                    "id": 27,
                    "instructions": "Bus towards Highlands Ranch",
                    "name": "South Broadway",
                    "num_stops": 11,
                    "possible_route_id": 7,
                    "short_name": "0",
                    "travel_mode": "TRANSIT",
                    "vehicle_type": "Bus",
                  },
                  Object {
                    "arrival_stop": null,
                    "arrival_time": null,
                    "color": null,
                    "credit_name": null,
                    "credit_url": null,
                    "departure_stop": null,
                    "departure_time": null,
                    "distance": "108 ft",
                    "duration": "1 min",
                    "headsign": null,
                    "id": 28,
                    "instructions": "Walk to 366 Broadway, Denver, CO 80203, USA",
                    "name": null,
                    "num_stops": null,
                    "possible_route_id": 7,
                    "short_name": null,
                    "travel_mode": "Walking",
                    "vehicle_type": null,
                  },
                ],
              },
            },
            "searchFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="favorites-div"
>
          <h2
                    className="favorites-title"
          >
                    Favorites
          </h2>
          <Favorite
                    deleteFavorite={[Function]}
                    favData={
                              Object {
                                        "0": Object {
                                          "arrival_time": "4:11pm",
                                          "departure_time": "3:51pm",
                                          "distance": "3.0 mi",
                                          "duration": "20 mins",
                                          "end_address": "366 Broadway, Denver, CO 80203, USA",
                                          "favorite": false,
                                          "itinerary_id": 7,
                                          "start_address": "Denver, CO 80202, USA",
                                          "steps": Array [
                                            Object {
                                              "arrival_stop": null,
                                              "arrival_time": null,
                                              "color": null,
                                              "credit_name": null,
                                              "credit_url": null,
                                              "departure_stop": null,
                                              "departure_time": null,
                                              "distance": "1 ft",
                                              "duration": "1 min",
                                              "headsign": null,
                                              "id": 26,
                                              "instructions": "Walk to Union Station",
                                              "name": null,
                                              "num_stops": null,
                                              "possible_route_id": 7,
                                              "short_name": null,
                                              "travel_mode": "Walking",
                                              "vehicle_type": null,
                                            },
                                            Object {
                                              "arrival_stop": "Broadway & 4th Ave",
                                              "arrival_time": "4:11pm",
                                              "color": "#0277bd",
                                              "credit_name": "Regional Transportation District",
                                              "credit_url": "http://rtd-denver.com/",
                                              "departure_stop": "Union Station",
                                              "departure_time": "3:52pm",
                                              "distance": "3.0 mi",
                                              "duration": "20 mins",
                                              "headsign": "Highlands Ranch",
                                              "id": 27,
                                              "instructions": "Bus towards Highlands Ranch",
                                              "name": "South Broadway",
                                              "num_stops": 11,
                                              "possible_route_id": 7,
                                              "short_name": "0",
                                              "travel_mode": "TRANSIT",
                                              "vehicle_type": "Bus",
                                            },
                                            Object {
                                              "arrival_stop": null,
                                              "arrival_time": null,
                                              "color": null,
                                              "credit_name": null,
                                              "credit_url": null,
                                              "departure_stop": null,
                                              "departure_time": null,
                                              "distance": "108 ft",
                                              "duration": "1 min",
                                              "headsign": null,
                                              "id": 28,
                                              "instructions": "Walk to 366 Broadway, Denver, CO 80203, USA",
                                              "name": null,
                                              "num_stops": null,
                                              "possible_route_id": 7,
                                              "short_name": null,
                                              "travel_mode": "Walking",
                                              "vehicle_type": null,
                                            },
                                          ],
                                        },
                                      }
                    }
                    searchFavorite={[Function]}
          />
</div>,
        "className": "favorite_card-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="favorites-title"
>
              Favorites
</h2>,
            Array [
              <Favorite
                deleteFavorite={[Function]}
                favData={
                                Object {
                                                "0": Object {
                                                  "arrival_time": "4:11pm",
                                                  "departure_time": "3:51pm",
                                                  "distance": "3.0 mi",
                                                  "duration": "20 mins",
                                                  "end_address": "366 Broadway, Denver, CO 80203, USA",
                                                  "favorite": false,
                                                  "itinerary_id": 7,
                                                  "start_address": "Denver, CO 80202, USA",
                                                  "steps": Array [
                                                    Object {
                                                      "arrival_stop": null,
                                                      "arrival_time": null,
                                                      "color": null,
                                                      "credit_name": null,
                                                      "credit_url": null,
                                                      "departure_stop": null,
                                                      "departure_time": null,
                                                      "distance": "1 ft",
                                                      "duration": "1 min",
                                                      "headsign": null,
                                                      "id": 26,
                                                      "instructions": "Walk to Union Station",
                                                      "name": null,
                                                      "num_stops": null,
                                                      "possible_route_id": 7,
                                                      "short_name": null,
                                                      "travel_mode": "Walking",
                                                      "vehicle_type": null,
                                                    },
                                                    Object {
                                                      "arrival_stop": "Broadway & 4th Ave",
                                                      "arrival_time": "4:11pm",
                                                      "color": "#0277bd",
                                                      "credit_name": "Regional Transportation District",
                                                      "credit_url": "http://rtd-denver.com/",
                                                      "departure_stop": "Union Station",
                                                      "departure_time": "3:52pm",
                                                      "distance": "3.0 mi",
                                                      "duration": "20 mins",
                                                      "headsign": "Highlands Ranch",
                                                      "id": 27,
                                                      "instructions": "Bus towards Highlands Ranch",
                                                      "name": "South Broadway",
                                                      "num_stops": 11,
                                                      "possible_route_id": 7,
                                                      "short_name": "0",
                                                      "travel_mode": "TRANSIT",
                                                      "vehicle_type": "Bus",
                                                    },
                                                    Object {
                                                      "arrival_stop": null,
                                                      "arrival_time": null,
                                                      "color": null,
                                                      "credit_name": null,
                                                      "credit_url": null,
                                                      "departure_stop": null,
                                                      "departure_time": null,
                                                      "distance": "108 ft",
                                                      "duration": "1 min",
                                                      "headsign": null,
                                                      "id": 28,
                                                      "instructions": "Walk to 366 Broadway, Denver, CO 80203, USA",
                                                      "name": null,
                                                      "num_stops": null,
                                                      "possible_route_id": 7,
                                                      "short_name": null,
                                                      "travel_mode": "Walking",
                                                      "vehicle_type": null,
                                                    },
                                                  ],
                                                },
                                              }
                }
                searchFavorite={[Function]}
/>,
            ],
          ],
          "className": "favorites-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Favorites",
              "className": "favorites-title",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "deleteFavorite": [Function],
              "favData": Object {
                "0": Object {
                  "arrival_time": "4:11pm",
                  "departure_time": "3:51pm",
                  "distance": "3.0 mi",
                  "duration": "20 mins",
                  "end_address": "366 Broadway, Denver, CO 80203, USA",
                  "favorite": false,
                  "itinerary_id": 7,
                  "start_address": "Denver, CO 80202, USA",
                  "steps": Array [
                    Object {
                      "arrival_stop": null,
                      "arrival_time": null,
                      "color": null,
                      "credit_name": null,
                      "credit_url": null,
                      "departure_stop": null,
                      "departure_time": null,
                      "distance": "1 ft",
                      "duration": "1 min",
                      "headsign": null,
                      "id": 26,
                      "instructions": "Walk to Union Station",
                      "name": null,
                      "num_stops": null,
                      "possible_route_id": 7,
                      "short_name": null,
                      "travel_mode": "Walking",
                      "vehicle_type": null,
                    },
                    Object {
                      "arrival_stop": "Broadway & 4th Ave",
                      "arrival_time": "4:11pm",
                      "color": "#0277bd",
                      "credit_name": "Regional Transportation District",
                      "credit_url": "http://rtd-denver.com/",
                      "departure_stop": "Union Station",
                      "departure_time": "3:52pm",
                      "distance": "3.0 mi",
                      "duration": "20 mins",
                      "headsign": "Highlands Ranch",
                      "id": 27,
                      "instructions": "Bus towards Highlands Ranch",
                      "name": "South Broadway",
                      "num_stops": 11,
                      "possible_route_id": 7,
                      "short_name": "0",
                      "travel_mode": "TRANSIT",
                      "vehicle_type": "Bus",
                    },
                    Object {
                      "arrival_stop": null,
                      "arrival_time": null,
                      "color": null,
                      "credit_name": null,
                      "credit_url": null,
                      "departure_stop": null,
                      "departure_time": null,
                      "distance": "108 ft",
                      "duration": "1 min",
                      "headsign": null,
                      "id": 28,
                      "instructions": "Walk to 366 Broadway, Denver, CO 80203, USA",
                      "name": null,
                      "num_stops": null,
                      "possible_route_id": 7,
                      "short_name": null,
                      "travel_mode": "Walking",
                      "vehicle_type": null,
                    },
                  ],
                },
              },
              "searchFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
